logstash-core/lib/logstash/agent_plugin_manager.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require "logstash/agent"
+
+module LogStash module AgentPluginManager
+  DEFAULT_AGENT_NAME = :default
+  class DuplicatePluginNameError < Error; end
+
+  REGISTRY = {}
+
+  # Reset plugin registry to just the default plugin
+  def self.reset!
+    REGISTRY.clear
+    REGISTRY[DEFAULT_AGENT_NAME] = LogStash::Agent
+  end
+
+  reset!
```

Comment:
Lets discuss about it, for sure. But we should definitely not do this kind of things as then errors are very complicated to find out. So yeah, look for another person to remove the conflict, but -1 from my side to this kind of constructs.